### PR TITLE
Refactor the port management code in ConfigurationManagementController

### PR DESCRIPTION
**Pull request by ptirador**
Related issue: #490 

Added `setPort` method, that allows to update a port passing it as a JSON in the request body.
Added entity `PortEntityBody` that represents a wrapper for the port argument.
Added `ResponseStatusEnum`, an enum that lists the plain text body messages (OK, FAILED).
All methods from `ConfigurationManagementController` have been reviewed to return the proper message according to its status code.

I also implemented the proper test cases.

### CODE REVIEW

**Comment by ptirador** [comment]
Refactored the base URL code in ConfigurationManagementController.

Requirement number 3 could not be implemented because in this case the path variable would be a URL and it wouldn't be very readable.
Added entity `BaseUrlEntityBody` that represents a wrapper for the base URL argument.

I also implemented the proper test cases.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java [hunk 1]
```diff
@@ -222,13 +240,14 @@ public ResponseEntity setProxyConfiguration(@ApiParam(value = "The storageId")
         try
         {
             configurationManagementService.setProxyConfiguration(storageId, repositoryId, proxyConfiguration);
-            return ResponseEntity.ok("The proxy configuration was updated successfully.");
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java]: Why have you removed the actual text response here?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java [hunk 2]
```diff
@@ -299,13 +318,14 @@ public ResponseEntity saveStorage(@ApiParam(value = "The storage object", requir
                 storageManagementService.createStorage(storage);
             }
 
-            return ResponseEntity.ok("The storage was updated successfully.");
+            return ResponseEntity.ok(ResponseStatusEnum.OK.value());
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java]: Why have you removed the actual text response here?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java [hunk 3]
```diff
@@ -368,14 +389,14 @@ public ResponseEntity removeStorage(@ApiParam(value = "The storageId", required
...
-                return ResponseEntity.ok("The storage was removed successfully.");
+                return ResponseEntity.ok(ResponseStatusEnum.OK.value());
             }
             catch (IOException | JAXBException e)
             {
                 logger.error(e.getMessage(), e);
 
                 return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
-                                     .body("Failed to remove storage " + storageId + "!");
+                                     .body(ResponseStatusEnum.FAILED.value());
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java]: Why have you removed the actual text response here?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java [hunk 4]
```diff
@@ -416,13 +437,14 @@ public ResponseEntity addOrUpdateRepository(@ApiParam(value = "The storageId", r
                 repositoryManagementService.createRepository(storageId, repository.getId());
             }
 
-            return ResponseEntity.ok("The repository was updated successfully.");
+            return ResponseEntity.ok(ResponseStatusEnum.OK.value());
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java]: Why have you removed the actual text response here?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java [hunk 5]
```diff
@@ -513,14 +535,14 @@ public ResponseEntity removeRepository(@ApiParam(value = "The storageId", requir
...
-                return ResponseEntity.ok().build();
+                return ResponseEntity.ok(ResponseStatusEnum.OK.value());
             }
             catch (IOException | JAXBException e)
             {
                 logger.error(e.getMessage(), e);
 
                 return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
-                                     .body("Failed to remove repository " + repositoryId + "!");
+                                     .body(ResponseStatusEnum.FAILED.value());
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/ConfigurationManagementController.java]: Why have you removed the actual text response here?

**Comment by ptirador** [comment]
Fixed `port` and `baseUrl` methods based on this guide:
[https://github.com/strongbox/strongbox/wiki/How-To-Implement-Spring-Controllers](url)

Tests also reviewed.

**Comment by steve-todorov** [comment]
@ptirador I've just tested your code and it works exactly as expected! Many thanks for your work on this! :)

@carlspring you can merge this once conflicts have been resolved.

**Comment by carlspring** [comment]
@ptirador ,

Could you just sort out he merge conflicts, so that we could merge it? Thanks a lot!

**Comment by ptirador** [comment]
Conflicts sorted out :+1:

**Comment by carlspring** [comment]
Thanks!

**Pull request merged**